doc/PerformanceTesting.md

```diff
@@ -0,0 +1,171 @@
+The perf framework will set an environment variable based on the service name with the value representing a version, thus allowing to run the tests against diffrent VCPKG published versions. If the env is not defined then the test will build against the current source code of the service. 
+There can be multiple set perf for each dependency of the service ( e.g. identity, storage). There is a bug in the main perf framework that causes some issues with multiple packages with different major versions, but it will get fixed.
+
+## Pipeline definition
+The file should be names perf.yml. or something similar, and should be located in the service directory that is to be tested. While not set in stone, since when creating a new pipeline in Azure we can point it an any place in the repo, it is recommended for consistency and to keep service related resources in the service folder. 
+
+```yml
+parameters:
+- name: PackageVersions
+  displayName: PackageVersions (regex of package versions to run)
+  type: string
+  default: '12|source'
+- name: Tests
+  displayName: Tests (regex of tests to run)
+  type: string
+  default: '^(download|upload|list-blobs)$'
+- name: Arguments
+  displayName: Arguments (regex of arguments to run)
+  type: string
+  default: '(10240)|(10485760)|(1073741824)|(5 )|(500 )|(50000 )' 
+- name: Iterations
+  displayName: Iterations (times to run each test)
+  type: number
+  default: '5'
+- name: AdditionalArguments
+  displayName: AdditionalArguments (passed to PerfAutomation)
+  type: string
+  default: ' '
+
+extends:
+  template: /eng/pipelines/templates/jobs/perf.yml
+  parameters:
+    ServiceDirectory: service folder
+    Services: "^service name$"
+    PackageVersions: ${{ parameters.PackageVersions }}
+    Tests: ${{ parameters.Tests }}
+    Arguments: ${{ parameters.Arguments }}
+    Iterations: ${{ parameters.Iterations }}
+    AdditionalArguments: ${{ parameters.AdditionalArguments }}
+    InstallLanguageSteps: 
+      - pwsh: |
+          Write-Host "##vso[task.setvariable variable=VCPKG_BINARY_SOURCES_SECRET;issecret=true;]clear;x-azblob,https://cppvcpkgcache.blob.core.windows.net/public-vcpkg-container,,read"
+        displayName: Set Vcpkg Variables
+
+    EnvVars: 
+      # This is set in the InstallLanguageSteps 
+      VCPKG_BINARY_SOURCES_SECRET: $(VCPKG_BINARY_SOURCES_SECRET)
+  
+```
+
+The fields of interest here are the parameters:
+- Tests which define which test s run in this pipeline ( which tests are available)
+- Arguments which allow to filter which subset of parameters are acceptable for the tests from the plurality in the framework
+- ServiceDirectory which represents the root folder of the service
+- Services which represents the services which these tests target. 
+  
+## Resources
+
+If the tests require certain resources to exist beforehand the pipeline can deploy them based on the presence of the **perf-resources.bicep** file which will be used to deploy the required resources defined within. 
```

Comment:
What if the test requires resources beyond those which can be expressed in a bicep file? Can the new-testresources.ps1 script be used instead of the bicep file?

Reply:
at the moment it uses bicep files, which are supported by azure https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/overview?tabs=bicep 

perf.yml -> /eng/common/TestResources/deploy-test-resources.yml -> eng/common/TestResources/New-TestResources.ps1 -- deploys bicep